src/main/java/io/cdap/plugin/cloud/vision/FilePathSourceConfig.java

```diff
@@ -0,0 +1,190 @@
+  @Macro
+  protected boolean recursive;
+
+  @Name(CloudVisionConstants.LAST_MODIFIED)
+  @Description("A way to filter files to be returned based on their last modified timestamp.")
+  @Macro
+  @Nullable
+  protected String lastModified;
+
+  @Name(CloudVisionConstants.SPLIT_BY)
+  @Description("Determines splitting mechanisms. Choose amongst default (uses the default splitting mechanism of " +
+    "file input format), directory (by each sub directory).")
+  @Macro
+  protected String splitBy;
+
+  public FilePathSourceConfig(String referenceName, String project, String serviceFilePath, String path,
+                              boolean recursive, String lastModified, String splitBy) {
+    this.referenceName = referenceName;
+    this.project = project;
+    this.serviceFilePath = serviceFilePath;
+    this.path = path;
+    this.recursive = recursive;
+    this.lastModified = lastModified;
+    this.splitBy = splitBy;
+  }
+
+  public String getReferenceName() {
+    return referenceName;
+  }
+
+  public String getPath() {
+    return path;
+  }
+
+  public boolean isRecursive() {
+    return recursive;
+  }
+
+  @Nullable
+  public String getLastModified() {
+    return lastModified;
+  }
+
+  @Nullable
+  public Long getLastModifiedEpochMilli() {
+    return Strings.isNullOrEmpty(lastModified) ? null : Instant.parse(lastModified).toEpochMilli();
+  }
+
+  public String getSplitBy() {
+    return splitBy;
+  }
+
+  public SplittingMechanism getSplittingMechanism() {
+    return Objects.requireNonNull(SplittingMechanism.fromDisplayName(splitBy));
+  }
+
+  public String getProject() {
+    String projectId = tryGetProject();
+    if (projectId == null) {
+      throw new IllegalArgumentException(
+        "Could not detect Google Cloud project id from the environment. Please specify a project id.");
+    }
+    return projectId;
+  }
+
+  @Nullable
+  public String tryGetProject() {
+    if (containsMacro(CloudVisionConstants.PROJECT) && Strings.isNullOrEmpty(project)) {
+      return null;
+    }
+    String projectId = project;
+    if (Strings.isNullOrEmpty(project) || CloudVisionConstants.AUTO_DETECT.equals(project)) {
+      projectId = ServiceOptions.getDefaultProjectId();
+    }
+    return projectId;
+  }
+
+  @Nullable
+  public String getServiceAccountFilePath() {
+    if (containsMacro(CloudVisionConstants.SERVICE_ACCOUNT_FILE_PATH) || Strings.isNullOrEmpty(serviceFilePath)
+      || CloudVisionConstants.AUTO_DETECT.equals(serviceFilePath)) {
+      return null;
+    }
+    return serviceFilePath;
+  }
+
+  /**
+   * Validates {@link FilePathSourceConfig} instance.
+   *
+   * @param collector failure collector.
+   */
+  public void validate(FailureCollector collector) {
+    if (Strings.isNullOrEmpty(referenceName)) {
+      collector.addFailure("Reference name must be specified", null)
+        .withConfigProperty(Constants.Reference.REFERENCE_NAME);
+    } else {
```

Comment:
`IdUtils.validateReferenceName(referenceName, collector);` does validation for empty String